modules/javafx.controls/src/main/java/javafx/scene/control/TableView.java

```diff
@@ -2642,7 +2642,7 @@ public void select(int row, TableColumn<S,?> column) {
         }
 
         @Override public void selectIndices(int row, int... rows) {
-            if (rows == null) {
+            if (rows == null || rows.length == 0) {
```

Comment:
To get `rows` to be `null` you'd have to call this method as: `selectIndices(2, null)` -- IMHO that should result in an exception, and not be accepted to be the same as `selectIndices(2)` or `selectIndices(2, new int[0])`.

I checked the documentation, and it does not mention that `null` is allowed here, which means it isn't.

Also: weird method signature, but I guess if you want to enforce at least one parameter it could be done this way.

Reply:
Well it's legal Java code, but that doesn't mean that leaving something `null` is allowed.  At the very least it is undocumented behavior:

    /**
     * <p>This method allows for one or more selections to be set at the same time.
     * It will ignore any value that is not within the valid range (i.e. greater
     * than or equal to zero, and less than the total number of items in the
     * underlying data model). Any duplication of indices will be ignored.
     *
     * <p>If there is already one or more indices selected in this model, calling
     * this method will <b>not</b> clear these selections - to do so it is
     * necessary to first call clearSelection.
     *
     * <p>The last valid value given will become the selected index / selected
     * item.
     * @param index the first index to select
     * @param indices zero or more additional indices to select
     */
    public abstract void selectIndices(int index, int... indices);

Reply:
I think it is also pretty clear the original author intended to check `rows.length == 0` and made the mistake that it would be called with `rows == null` when there are no further indices specified, which is incorrect.

Reply:
I also think that `null` should throw unless otherwise specified.

Reply:
Since the null check is preexisting, I don't think we should make the change to throw NPE on null as part of a simple bug fix. If we decide to change this, I would want to see a spec change in the docs with an accompanying CSR, which should be done as a separate bug.